### PR TITLE
SDK-1326 [Android] Collect, Persist, and Forward 'gclid' value

### PR DESCRIPTION
**Pull request by gdeluna-branch**
## Reference
SDK-1326 -- [Android] Collect, Persist, and Forward 'gclid' value.

## Description
This PR collects the `gclid` from install referrer extras and persists them in SharedPreferences.

The value is sent on all events (except install) after being set in a new helper function updateRequestWithParamsAllEvents.
The gclid value remains stored locally along with the time it was written in a json object. When attempting to read the gclid value, we evaluate if it's still valid as computed with the validity window. 

That window is set by default to be 30 days, the default can be overridden by calling setReferrerGclidValidForWindow, for example:

        Branch.getInstance().setReferrerGclidValidForWindow(1000000L);

## Testing Instructions
Manual testing with test install referrer extra data in a test app.

`referrer_gclid` is a top level field in both v1 (except install) and v2 events 

## Risk Assessment [`LOW`]

- [✅ ] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- [ ] JIRA Ticket is referenced in PR title.
- Correctness & Style
    - [ ] Conforms to [AOSP Style Guides](https://source.android.com/setup/contribute/code-style)
    - [ ] Mission critical pieces are documented in code and out of code as needed.
- [ ] Unit Tests reviewed and test issue sufficiently.
- [ ] Functionality was reviewed in QA independently by another engineer on the team.

### CODE REVIEW

File: Branch-SDK/src/main/java/io/branch/referral/DeviceInfo.java [hunk 1]
```diff
@@ -100,7 +101,9 @@ void updateRequestWithV1Params(ServerRequest serverRequest, JSONObject requestOb
                 requestObj.put(Defines.Jsonkey.LocalIP.getKey(), localIpAddr);
             }
 
-            if (PrefHelper.getInstance(context_).shouldAddModules()) {
+            PrefHelper prefHelper_ = PrefHelper.getInstance(context_);
```
Comment (gdeluna-branch) [inline Branch-SDK/src/main/java/io/branch/referral/DeviceInfo.java:104]: undo unused change

File: Branch-SDK/src/main/java/io/branch/referral/PrefHelper.java [hunk 2]
```diff
@@ -632,8 +637,72 @@ public void setGooglePlayReferrer(String referrer) {
...
+        }
+
+        return gclid;
+    }
+
+    /**
+     * Sets the GCLID expiry window in milliseconds
+     * @param window
+     */
+    public void setReferrerGclidValidForWindow(long window){
```
Comment (gdeluna-branch) [inline Branch-SDK/src/main/java/io/branch/referral/PrefHelper.java:694]: validate input - e.g not in the past

File: Branch-SDK/src/main/java/io/branch/referral/PrefHelper.java [hunk 3]
```diff
@@ -632,8 +637,72 @@ public void setGooglePlayReferrer(String referrer) {
...
+            // If expiry time has not elapsed, return it
+            if(expiryDate - System.currentTimeMillis() > 0){
+                gclid = gclidJsonObject.getString(KEY_GCLID_VALUE);
+            }
+            // Else delete it
+            else{
+                removePrefValue(KEY_GCLID_JSON_OBJECT);
+            }
+        } catch (JSONException e) {
+            e.printStackTrace();
```
Comment (gdeluna-branch) [inline Branch-SDK/src/main/java/io/branch/referral/PrefHelper.java:684]: If there's a malformed json - might as well remove it

File: Branch-SDK/src/androidTest/java/io/branch/referral/PrefHelperTest.java [hunk 4]
```diff
@@ -101,4 +101,42 @@ public void testSetTaskTimeout(){
         int result = prefHelper.getTaskTimeout();
         Assert.assertEquals(TEST_TIMEOUT + TEST_CONNECT_TIMEOUT, result);
     }
+
+    @Test
+    public void testSetReferrerGclidValidForWindow(){
```
Comment (gdeluna-branch) [inline Branch-SDK/src/androidTest/java/io/branch/referral/PrefHelperTest.java:106]: add over/underflow tests

**Review by agerstner-branch — APPROVED** [review]
LGTM

**Pull request merged**